Update Upstream (SAPPHIRE)

Dueris opened the pull request.

D3st0ny committed:
fix stasis chamber behaviour

Dueris committed:
Merge pull request #129 from D3st0ny/ver/1.21.8

fix stasis chamber behaviour

Dueris committed:
Ensure goal isn't null

Dueris committed:
Slight oopsies in region threading fix

Dueris committed:
More region threading fixes

Dueris committed:
Make asynchronous protocol switch configurable

Dueris committed:
Update Upstream (Paper/Folia)

Suisuroru reviewed (commented):
I think you should respect the origin author, you are violating our open source license.

related commit: https://github.com/CraftCanvasMC/Canvas/pull/130/commits/9510dfcd1a243a25ef80aa58e20b6c7f171ca893

origin changes: https://github.com/LuminolMC/Luminol/blob/ver/1.21.8/luminol-server/minecraft-patches/features/0062-Fix-off-tickregion-sync-teleport.patch  and https://github.com/LuminolMC/Luminol/blob/ver/1.21.8/luminol-server/minecraft-patches/features/0055-Correct-player-respawn-place.patch

Dueris commented:
Fixed in https://github.com/CraftCanvasMC/Canvas/pull/131
Apologies for the inconvenience, those 2 changes were built from a patch file sent to me by a private fork developer who I was discussing with. They run a fork of Canvas for their server and included patches of their own and sent them to me to be downstreamed. The changes were all unmarked(no "// blah start" and such), so I added them myself. Was not intentional.
